File: backend/Spotly/Spotly/Models/Zahtjev.cs
using System;
using System.Collections.Generic;

namespace Spotly.Models;

public partial class Zahtjev
{
    public int Id { get; set; }

    public string Predmet { get; set; } = null!;

    public string Poruka { get; set; } = null!;

    public string? Odgovor { get; set; }

    public string? Status { get; set; }

    public byte[] DatumVrijeme { get; set; } = null!;

    public int? AdminId { get; set; }

    public int? KorisnikId { get; set; }

    public int? TipZahtjevaId { get; set; }

    public virtual ICollection<Dokumentacija> Dokumentacijas { get; set; } = new List<Dokumentacija>();

    public virtual Korisnik? Korisnik { get; set; }

    public virtual TipZahtjeva? TipZahtjeva { get; set; }
}

File: frontend/app/src/main/java/org/foi/hr/air/spotly/data/Zahtjev.kt
package org.foi.hr.air.spotly.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class Zahtjev(
    @SerialName("id") val id: Int,
    @SerialName("predmet") val predmet: String,
    @SerialName("poruka") val poruka: String,
    @SerialName("odgovor") val odgovor: String?,
    @SerialName("status") val status: String,
    @SerialName("datumVrijeme") val datumVrijeme: String,
    @SerialName("adminId") val adminId: Int?,
    @SerialName("korisnikId") val korisnikId: Int,
    @SerialName("tipZahtjevaId") val tipZahtjevaId: Int?
)
